### PR TITLE
GEOPY-287: run pre-commit update - followup

### PR DESCRIPTION
**Pull request by sebhmg**
**GEOPY-287 - apply pre-commit update** 
 followup of https://github.com/MiraGeoscience/geoh5py/pull/107

### CODE REVIEW

File: geoh5py/workspace/workspace.py [hunk 1]
```diff
@@ -626,9 +626,9 @@ def finalize(self, file: str | h5py.File | None = None):
         :param file: :obj:`h5py.File` or name of the target geoh5 file
         """
         for entity in (
-            cast(List["Entity"], self.objects)
-            + cast(List["Entity"], self.groups)
-            + cast(List["Entity"], self.data)
+            cast(list[Entity], self.objects)  # pylint: disable=unsubscriptable-object
```
Comment (sebhmg) [inline geoh5py/workspace/workspace.py:628]: although cast is ignored at runtime, `pylint` does not like the new type hint syntax there

File: geoh5py/workspace/workspace.py [hunk 2]
```diff
@@ -120,22 +120,22 @@ def active() -> Workspace:
     def _all_data(self) -> list[data.Data]:
         """Get all active Data entities registered in the workspace."""
         self.remove_none_referents(self._data, "Data")
-        return [cast("data.Data", v()) for v in self._data.values()]
+        return [cast(data.Data, v()) for v in self._data.values()]
```
Comment (sebhmg) [inline geoh5py/workspace/workspace.py:122]: using string was known as forward declaration
Not needed anymore thanks too [PEP-563](https://www.python.org/dev/peps/pep-0563/)

**Comment by sebhmg** [comment]
pre-commit.ci run

**Review by domfournier — APPROVED** [review]
Very nice.
image

**Comment by sebhmg** [comment]
pre-commit.ci run

**Pull request merged**